### PR TITLE
Fix issues with Bloop snapshot versions

### PR DESCRIPTION
**Pull request by tgodzik**

### CODE REVIEW

**Review by kpodsiad — APPROVED** [review]
I wonder if using https://github.com/vdurmont/semver4j wouldn't be simpler :P

File: mtags/src/main/scala/scala/meta/internal/semver/SemVer.scala [hunk 1]
```diff
@@ -31,7 +31,7 @@ object SemVer {
           .getOrElse(List(2, 0))
 
       List(major, minor, patch) ++ rcMilestonePart ++
-        List(nightlyDate.getOrElse(Int.MaxValue))
+        List(nightlyDate.getOrElse(0))
```
Comment (dos65) [inline mtags/src/main/scala/scala/meta/internal/semver/SemVer.scala:34]: For scala versions it should be `Int.MaxValue`.
There is a difference with bloop versions: 
- for scala nightly `3.0.0` is greater than `3.0.0-bin-20220310-NIGHTLY`
- for bloop  `1.4.13` is less than `1.4.13-$n-$sha-20220310`

To make it work correctly we either need to introduce a different version impl for bloop or change bloop versioning scheme.

Comment (tgodzik) [reply mtags/src/main/scala/scala/meta/internal/semver/SemVer.scala:34]: Maybe we should just detect if it ends with nightly?

Comment (dos65) [reply mtags/src/main/scala/scala/meta/internal/semver/SemVer.scala:34]: @tgodzik maybe the right option would be to revert SemVer.Version back and introduce a specific ScalaVersion?
I just realized that scala-nightly version doesn't follow semver.

Comment (dos65) [reply mtags/src/main/scala/scala/meta/internal/semver/SemVer.scala:34]: > Maybe we should just detect if it ends with nightly?

The initial motivation of this change was to able to compare NIGHTLY version between each other too.

Comment (tgodzik) [reply mtags/src/main/scala/scala/meta/internal/semver/SemVer.scala]: It should work correctly now, I added a check for `NIGHTLY`, which we recongnize as being only the scala nightly

**Review by dos65 — APPROVED** [review]
Looks good! Thx for the fix!

**Pull request merged**